src/snowflake/cli/app/dev/docs/project_definition_generate_json_schema.py

```diff
@@ -0,0 +1,219 @@
+                path=field_name,
+                title=field_model["title"],
+                indents=0,
+                item_index=0,
+                is_required=is_required,
+                field_name=field_name,
+                add_types=len(children_fields) == 0,
+                types=" | ".join(self._get_field_types(field_model)),
+            )
+            fields = [new_field] + children_fields
+
+            if is_required:
+                required_fields += fields
+            else:
+                sections += [
+                    {
+                        "fields": fields,
+                        "title": field_model["title"],
+                        "name": field_name,
+                    }
+                ]
+
+        for section in sections:
+            section["fields"] = required_fields + section["fields"]
+
+        return sections
+
+    def _get_section_fields(
+        self,
+        current_definition: Dict[str, Any],
+        current_path: str = "",
+        deep: int = 0,
+        is_array_item: bool = False,
+    ) -> List[Dict[str, Any]]:
+        required_fields: List[Dict[str, Any]] = []
+        optional_fields: List[Dict[str, Any]] = []
+        item_index = 0
+
+        for field_name, field_model in current_definition["properties"].items():
+            item_index += 1 if is_array_item else 0
+            is_required = (
+                "required" in current_definition
+                and field_name in current_definition["required"]
+            )
+            new_current_path = (
+                field_name if current_path == "" else current_path + "." + field_name
+            )
+            children_fields = self._get_children_fields(
+                field_model, new_current_path, deep
+            )
+            new_field = self._create_new_field(
+                path=new_current_path,
+                title=field_model["title"],
+                indents=deep,
+                item_index=item_index,
+                is_required=is_required,
+                field_name=field_name,
+                add_types=len(children_fields) == 0,
+                types=" | ".join(self._get_field_types(field_model)),
+            )
+            fields = [new_field] + children_fields
+            if is_required:
+                required_fields += fields
+            else:
+                optional_fields += fields
+        return required_fields + optional_fields
+
+    def _create_new_field(
+        self,
+        path: str,
+        title: str,
+        indents: int,
+        item_index: int,
+        is_required: bool,
+        field_name: str,
+        add_types: bool,
+        types: str,
+    ):
+        return {
+            "path": path,
+            "title": title,
+            "indents": indents,
+            "item_index": item_index,
+            "required": is_required,
+            "name": field_name,
+            "add_types": add_types,
+            "types": types,
+        }
```

Comment:
We can consider introducing a dataclass for those information. Later we can do `asdict(cls)`

Reply:
I think it is not necessary since it doesn't add more information to the developer, it will not be used in any other place in the code and it add an extra step for the same result

Reply:
Imho it will reduce the duplication and will make the code cleaner

Reply:
Is true, it looks cleaner. Thank you!